community/community-it/community-it/src/test/java/org/neo4j/io/pagecache/impl/muninn/PageListTest.java

```diff
@@ -141,504 +133,699 @@ public void mustExposePageCount()
     {
+        init( pageId );
+
         long r = pageList.tryOptimisticReadLock( pageRef );
         pageList.tryWriteLock( pageRef );
         assertFalse( pageList.validateReadLock( pageRef, r ) );
     }
 
-    @Test
-    public void optimisticReadLockMustNotValidateUnderWriteLock()
+    @ParameterizedTest( name = "pageRef = {0}" )
+    @MethodSource( "argumentsProvider" )
+    public void optimisticReadLockMustNotValidateUnderWriteLock( int pageId )
     {
+        init( pageId );
+
         pageList.tryWriteLock( pageRef );
         long r = pageList.tryOptimisticReadLock( pageRef );
         assertFalse( pageList.validateReadLock( pageRef, r ) );
     }
 
-    @Test
-    public void writeLockReleaseMustInvalidateOptimisticReadLock()
+    @ParameterizedTest( name = "pageRef = {0}" )
+    @MethodSource( "argumentsProvider" )
+    public void writeLockReleaseMustInvalidateOptimisticReadLock( int pageId )
     {
+        init( pageId );
+
         pageList.unlockExclusive( pageRef );
         pageList.tryWriteLock( pageRef );
         long r = pageList.tryOptimisticReadLock( pageRef );
         pageList.unlockWrite( pageRef );
         assertFalse( pageList.validateReadLock( pageRef, r ) );
     }
 
-    @Test
-    public void uncontendedWriteLockMustBeAvailable()
+    @ParameterizedTest( name = "pageRef = {0}" )
+    @MethodSource( "argumentsProvider" )
+    public void uncontendedWriteLockMustBeAvailable( int pageId )
     {
+        init( pageId );
+
         pageList.unlockExclusive( pageRef );
         assertTrue( pageList.tryWriteLock( pageRef ) );
     }
 
-    @Test
-    public void uncontendedOptimisticReadLockMustValidateAfterWriteLockRelease()
+    @ParameterizedTest( name = "pageRef = {0}" )
+    @MethodSource( "argumentsProvider" )
+    public void uncontendedOptimisticReadLockMustValidateAfterWriteLockRelease( int pageId )
     {
+        init( pageId );
+
         pageList.unlockExclusive( pageRef );
         pageList.tryWriteLock( pageRef );
         pageList.unlockWrite( pageRef );
         long r = pageList.tryOptimisticReadLock( pageRef );
         assertTrue( pageList.validateReadLock( pageRef, r ) );
     }
 
-    @Test( timeout = TIMEOUT )
-    public void writeLocksMustNotBlockOtherWriteLocks()
+    @ParameterizedTest( name = "pageRef = {0}" )
+    @MethodSource( "argumentsProvider" )
+    public void writeLocksMustNotBlockOtherWriteLocks( int pageId )
     {
-        pageList.unlockExclusive( pageRef );
-        assertTrue( pageList.tryWriteLock( pageRef ) );
-        assertTrue( pageList.tryWriteLock( pageRef ) );
-    }
+        init( pageId );
 
-    @Test( timeout = TIMEOUT )
-    public void writeLocksMustNotBlockOtherWriteLocksInOtherThreads() throws Exception
-    {
-        pageList.unlockExclusive( pageRef );
-        int threads = 10;
-        CountDownLatch end = new CountDownLatch( threads );
-        Runnable runnable = () ->
+        assertTimeout( ofSeconds( 5 ), () ->
```

Comment:
Here you should use `assertTimeoutPreemptively`. `assertTimeout` is executed in the same thread, so in case of a deadlock, that test will not be killed after the specified timeout.

I've fixed it already so you don't have to address it.

Reply:
Thanks for info, I didn't know that.

I guess the conflict should also be resolved, let me know if I should do it.

Reply:
No, it's fine, I did a quick resolve when I cherry-picked it. CI is running now.